File: ios/App/View/Navigation/SidebarMenuView.swift
import SwiftUI

/// Drill-up affordance rendered above the nav when a hierarchical app is
/// inside a child nav level (see ShellNavLevel).
struct ShellDrillUp {
  let label: String
  let action: () -> Void
}

/// An app-supplied entry in the account menu (e.g. "Account settings").
struct ShellProfileMenuItem: Identifiable {
  let id: String
  let label: String
  var systemImage: String? = nil
  let action: () -> Void
}

/// The signed-in account shown in the drawer footer.
struct ShellProfile {
  let label: String
  var sublabel: String? = nil
  /// Avatar image URL; the label's initials render when nil (web twin:
  /// AppShellProfile.avatarUrl).
  var avatarURL: URL? = nil
  /// Rendered above the built-in theme toggle + sign out rows; what the
  /// items do is the binder's concern (the shell stays domain-agnostic).
  var items: [ShellProfileMenuItem] = []
  var onSignOut: (() -> Void)? = nil
}

/// The sidebar drawer: brand row, sectioned nav (badges, expandable
/// children, drill-up), and the account footer with theme + sign out —
/// the native twin of the web design-system AppShell sidebar. Purely
/// presentational; nav data and handlers are injected.
struct SidebarMenuView: View {
  @Environment(\.uiThemeTokens) private var theme
  @AppStorage("base.themeMode") private var themeModeSetting = "system"

  let title: String
  let sections: [ShellNavSection]
  var drillUp: ShellDrillUp? = nil
  var profile: ShellProfile? = nil
  let selectedItemId: String
  let onSelect: (ShellNavItem) -> Void

  @State private var expandedGroupIds: Set<String> = []
  @State private var isAccountMenuOpen = false

  var body: some View {
    VStack(alignment: .leading, spacing: 0) {
      brandRow

      ScrollView {
        VStack(alignment: .leading, spacing: theme.spacing.md) {
          if let drillUp {
            drillUpRow(drillUp)
          }
          ForEach(Array(sections.enumerated()), id: \.element.id) { index, section in
            sectionView(section, isFirst: index == 0)
          }
        }
        .padding(.horizontal, theme.spacing.sm)
        .padding(.vertical, theme.spacing.sm)
      }

      if let profile {
        accountFooter(profile)
      }
    }
    .frame(maxHeight: .infinity, alignment: .top)
    .background(theme.colors.surface)
    .overlay(alignment: .trailing) {
      Rectangle()
        .fill(theme.colors.border)
        .frame(width: 1)
    }
    .onAppear {
      expandGroupContainingSelection()
    }
    .onChange(of: selectedItemId) {
      expandGroupContainingSelection()
    }
  }

  private var brandRow: some View {
    HStack(spacing: theme.spacing.sm) {
      Text(title)
        .font(theme.typography.font(size: theme.typography.sizes.lg, weight: .bold))
        .foregroundStyle(theme.colors.textPrimary)
        .lineLimit(1)
      Spacer(minLength: 0)
    }
    .padding(.horizontal, theme.spacing.md)
    .padding(.vertical, theme.spacing.md)
  }

  private func drillUpRow(_ drillUp: ShellDrillUp) -> some View {
    Button(action: drillUp.action) {
      HStack(spacing: theme.spacing.sm) {
        Image(systemName: "chevron.backward")
          .font(.system(size: theme.typography.sizes.sm, weight: .semibold))
        Text(drillUp.label)
          .font(theme.typography.font(size: theme.typography.sizes.sm, weight: .semibold))
        Spacer(minLength: 0)
      }
      .foregroundStyle(theme.colors.textSecondary)
      .padding(.horizontal, theme.spacing.sm)
      .padding(.vertical, theme.spacing.xs)
    }
    .buttonStyle(.plain)
  }

  @ViewBuilder
  private func sectionView(_ section: ShellNavSection, isFirst: Bool) -> some View {
    VStack(alignment: .leading, spacing: theme.spacing.xxs) {
      if let sectionTitle = section.title {
        Text(sectionTitle.uppercased())
          .font(theme.typography.font(size: theme.typography.sizes.xs, weight: .bold))
          .kerning(0.8)
          .foregroundStyle(theme.colors.textSecondary)
          .padding(.horizontal, theme.spacing.sm)
          .padding(.top, theme.spacing.xs)
      } else if !isFirst {
        Rectangle()
          .fill(theme.colors.border)
          .frame(height: 1)
          .padding(.horizontal, theme.spacing.sm)
          .padding(.vertical, theme.spacing.xs)
      }
      ForEach(section.items) { item in
        navRow(item, isChild: false)
        if !item.children.isEmpty, expandedGroupIds.contains(item.id) {
          ForEach(item.children) { child in
            navRow(child, isChild: true)
          }
        }
      }
    }
  }

  @ViewBuilder
  private func navRow(_ item: ShellNavItem, isChild: Bool) -> some View {
    let isActive = selectedItemId == item.id
    let hasChildren = !item.children.isEmpty
    Button {
      if hasChildren {
        toggleGroup(item.id)
      } else {
        onSelect(item)
      }
    } label: {
      HStack(spacing: theme.spacing.sm) {
        if let systemImage = item.systemImage {
          Image(systemName: systemImage)
            .font(.system(size: theme.typography.sizes.md, weight: .medium))
            .frame(width: 20)
        }
        Text(item.label)
          .font(theme.typography.font(size: theme.typography.sizes.sm, weight: .semibold))
          .lineLimit(1)
        Spacer(minLength: 0)
        if let badgeText = item.badgeText, !badgeText.isEmpty, badgeText != "0" {
          Text(badgeText)
            .font(theme.typography.font(size: theme.typography.sizes.xs, weight: .bold))
            .foregroundStyle(theme.colors.accent)
            .padding(.horizontal, theme.spacing.xs)
            .padding(.vertical, 1)
            .background(Capsule().fill(theme.colors.accent.opacity(0.14)))
        }
        if hasChildren {
          Image(systemName: "chevron.forward")
            .font(.system(size: theme.typography.sizes.xs, weight: .semibold))
            .foregroundStyle(theme.colors.textSecondary)
            .rotationEffect(.degrees(expandedGroupIds.contains(item.id) ? 90 : 0))
        }
      }
      .foregroundStyle(isActive ? theme.colors.accent : theme.colors.textSecondary)
      .padding(.horizontal, theme.spacing.sm)
      .padding(.vertical, theme.spacing.xs + 2)
      .padding(.leading, isChild ? theme.spacing.xl : 0)
      .background(
        RoundedRectangle(cornerRadius: theme.radius.md, style: .continuous)
          .fill(isActive ? theme.colors.accent.opacity(0.12) : Color.clear)
      )
      .contentShape(Rectangle())
    }
    .buttonStyle(.plain)
    .accessibilityLabel(item.label)
  }

  @ViewBuilder
  private func accountFooter(_ profile: ShellProfile) -> some View {
    VStack(alignment: .leading, spacing: theme.spacing.xs) {
      Rectangle()
        .fill(theme.colors.border)
        .frame(height: 1)

      if isAccountMenuOpen {
        ForEach(profile.items) { item in
          Button(action: item.action) {
            menuRow(systemImage: item.systemImage ?? "gearshape", label: item.label)
          }
          .buttonStyle(.plain)
        }

        Button {
          themeModeSetting = theme.mode == .light ? "dark" : "light"
        } label: {
          menuRow(
            systemImage: theme.mode == .light ? "moon" : "sun.max",
            label: theme.mode == .light ? "Dark mode" : "Light mode"
          )
        }
        .buttonStyle(.plain)

        if let onSignOut = profile.onSignOut {
          Button(action: onSignOut) {
            menuRow(systemImage: "rectangle.portrait.and.arrow.right", label: "Sign out")
          }
          .buttonStyle(.plain)
        }
      }

      Button {
        withAnimation(.easeInOut(duration: 0.15)) {
          isAccountMenuOpen.toggle()
        }
      } label: {
        HStack(spacing: theme.spacing.sm) {
          profileAvatar(profile)
          VStack(alignment: .leading, spacing: 1) {
            Text(profile.label)
              .font(theme.typography.font(size: theme.typography.sizes.sm, weight: .semibold))
              .foregroundStyle(theme.colors.textPrimary)
              .lineLimit(1)
            if let sublabel = profile.sublabel {
              Text(sublabel)
                .font(theme.typography.font(size: theme.typography.sizes.xs))
                .foregroundStyle(theme.colors.textSecondary)
                .lineLimit(1)
            }
          }
          Spacer(minLength: 0)
          Image(systemName: "chevron.up")
            .font(.system(size: theme.typography.sizes.xs, weight: .semibold))
            .foregroundStyle(theme.colors.textSecondary)
            .rotationEffect(.degrees(isAccountMenuOpen ? 180 : 0))
        }
        .padding(.horizontal, theme.spacing.sm)
        .padding(.vertical, theme.spacing.xs)
        .contentShape(Rectangle())
      }
      .buttonStyle(.plain)
      .accessibilityLabel("Account: \(profile.label)")
    }
    .padding(.horizontal, theme.spacing.sm)
    .padding(.bottom, theme.spacing.sm)
  }

  /// The footer's avatar affordance: the uploaded avatar image when the
  /// profile carries one (storage-kernel PUBLIC serving URL), otherwise the
  /// label's initials — matching the web AppShell's avatar rendering.
  @ViewBuilder
  private func profileAvatar(_ profile: ShellProfile) -> some View {
    if let avatarURL = profile.avatarURL {
      AsyncImage(url: avatarURL) { image in
        image
          .resizable()
          .scaledToFill()
      } placeholder: {
        profileInitials(profile.label)
      }
      .frame(width: 28, height: 28)
      .clipShape(Circle())
    } else {
      profileInitials(profile.label)
    }
  }

  private func profileInitials(_ label: String) -> some View {
    Text(Self.initials(of: label))
      .font(theme.typography.font(size: theme.typography.sizes.xs, weight: .bold))
      .foregroundStyle(theme.colors.accent)
      .frame(width: 28, height: 28)
      .background(Circle().fill(theme.colors.accent.opacity(0.14)))
  }

  private func menuRow(systemImage: String, label: String) -> some View {
    HStack(spacing: theme.spacing.sm) {
      Image(systemName: systemImage)
        .font(.system(size: theme.typography.sizes.sm, weight: .medium))
        .frame(width: 20)
      Text(label)
        .font(theme.typography.font(size: theme.typography.sizes.sm, weight: .semibold))
      Spacer(minLength: 0)
    }
    .foregroundStyle(theme.colors.textPrimary)
    .padding(.horizontal, theme.spacing.sm)
    .padding(.vertical, theme.spacing.xs + 2)
    .contentShape(Rectangle())
  }

  private func toggleGroup(_ id: String) {
    withAnimation(.easeInOut(duration: 0.15)) {
      if expandedGroupIds.contains(id) {
        expandedGroupIds.remove(id)
      } else {
        expandedGroupIds.insert(id)
      }
    }
  }

  private func expandGroupContainingSelection() {
    for section in sections {
      for item in section.items where item.children.contains(where: { $0.id == selectedItemId }) {
        expandedGroupIds.insert(item.id)
      }
    }
  }

  static func initials(of label: String) -> String {
    let parts = label.split(whereSeparator: { " @._-".contains($0) }).filter { !$0.isEmpty }
    guard let first = parts.first else { return "?" }
    if parts.count == 1 {
      return String(first.prefix(2)).uppercased()
    }
    return String([first.first!, parts[1].first!]).uppercased()
  }
}

File: android/app/src/main/kotlin/com/baseapp/android/view/navigation/SidebarMenuView.kt
package com.baseapp.android.view.navigation

import androidx.compose.animation.AnimatedVisibility
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxHeight
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.statusBarsPadding
import androidx.compose.foundation.layout.navigationBarsPadding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.Logout
import androidx.compose.material.icons.filled.ChevronLeft
import androidx.compose.material.icons.filled.DarkMode
import androidx.compose.material.icons.filled.KeyboardArrowDown
import androidx.compose.material.icons.filled.KeyboardArrowUp
import androidx.compose.material.icons.filled.LightMode
import androidx.compose.material.icons.filled.Settings
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.rotate
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.baseapp.android.components.shell.ShellNavItem
import com.baseapp.android.components.shell.ShellNavSection
import com.baseapp.android.view.kit.RemoteAvatarImage
import com.baseapp.android.view.theme.LocalUiTheme
import com.baseapp.android.view.theme.ThemeModeStore
import com.baseapp.android.view.theme.UiThemeMode
import java.util.Locale

/** Drill-up affordance rendered above the nav when inside a child nav level. */
data class ShellDrillUp(
    val label: String,
    val action: () -> Unit,
)

/** An app-supplied entry in the account menu (e.g. "Account settings"). */
data class ShellProfileMenuItem(
    val id: String,
    val label: String,
    val icon: ImageVector? = null,
    val action: () -> Unit,
)

/** The signed-in account shown in the drawer footer. */
data class ShellProfile(
    val label: String,
    val sublabel: String? = null,
    /** PUBLIC avatar upload's serving URL; initials render when null. */
    val avatarUrl: String? = null,
    /**
     * Rendered above the built-in theme toggle + sign out rows; what the
     * items do is the binder's concern (the shell stays domain-agnostic).
     */
    val items: List<ShellProfileMenuItem> = emptyList(),
    val onSignOut: (() -> Unit)? = null,
)

/**
 * The sidebar drawer: brand row, sectioned nav (badges, expandable children,
 * drill-up), and the account footer with theme + sign out — the native twin
 * of the web design-system AppShell sidebar. Purely presentational; nav data
 * and handlers are injected. Icons come from the binder via [iconForItemId].
 */
@Composable
fun SidebarMenuView(
    title: String,
    sections: List<ShellNavSection>,
    selectedItemId: String,
    onSelect: (ShellNavItem) -> Unit,
    drillUp: ShellDrillUp? = null,
    profile: ShellProfile? = null,
    iconForItemId: Map<String, ImageVector> = emptyMap(),
) {
    val theme = LocalUiTheme.current
    val expandedGroupIds = remember {
        mutableStateOf(
            sections
                .flatMap { it.items }
                .filter { item -> item.children.any { it.id == selectedItemId } }
                .map { it.id }
                .toSet(),
        )
    }

    Column(
        modifier = Modifier
            .fillMaxHeight()
            .background(theme.colors.surface)
            .statusBarsPadding()
            .navigationBarsPadding(),
    ) {
        Text(
            text = title,
            color = theme.colors.textPrimary,
            fontSize = theme.typography.sizes.lg,
            fontWeight = FontWeight.Bold,
            maxLines = 1,
            modifier = Modifier.padding(horizontal = theme.spacing.md, vertical = theme.spacing.md),
        )

        Column(
            modifier = Modifier
                .weight(1f)
                .verticalScroll(rememberScrollState())
                .padding(horizontal = theme.spacing.sm),
        ) {
            drillUp?.let { DrillUpRow(it) }
            sections.forEachIndexed { index, section ->
                SectionView(
                    section = section,
                    isFirst = index == 0,
                    selectedItemId = selectedItemId,
                    expandedGroupIds = expandedGroupIds.value,
                    onToggleGroup = { id ->
                        expandedGroupIds.value =
                            if (id in expandedGroupIds.value) expandedGroupIds.value - id
                            else expandedGroupIds.value + id
                    },
                    onSelect = onSelect,
                    iconForItemId = iconForItemId,
                )
            }
        }

        profile?.let { AccountFooter(it) }
    }
}

@Composable
private fun DrillUpRow(drillUp: ShellDrillUp) {
    val theme = LocalUiTheme.current
    Row(
        verticalAlignment = Alignment.CenterVertically,
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(theme.radius.md))
            .clickable(onClick = drillUp.action)
            .padding(horizontal = theme.spacing.sm, vertical = theme.spacing.xs),
    ) {
        Icon(
            imageVector = Icons.Filled.ChevronLeft,
            contentDescription = null,
            tint = theme.colors.textSecondary,
            modifier = Modifier.size(18.dp),
        )
        Text(
            text = drillUp.label,
            color = theme.colors.textSecondary,
            fontSize = theme.typography.sizes.sm,
            fontWeight = FontWeight.SemiBold,
            modifier = Modifier.padding(start = theme.spacing.sm),
        )
    }
}

@Composable
private fun SectionView(
    section: ShellNavSection,
    isFirst: Boolean,
    selectedItemId: String,
    expandedGroupIds: Set<String>,
    onToggleGroup: (String) -> Unit,
    onSelect: (ShellNavItem) -> Unit,
    iconForItemId: Map<String, ImageVector>,
) {
    val theme = LocalUiTheme.current
    val sectionTitle = section.title
    if (sectionTitle != null) {
        Text(
            text = sectionTitle.uppercase(Locale.getDefault()),
            color = theme.colors.textSecondary,
            fontSize = theme.typography.sizes.xs,
            fontWeight = FontWeight.Bold,
            modifier = Modifier.padding(
                start = theme.spacing.sm,
                top = theme.spacing.md,
                bottom = theme.spacing.xxs,
            ),
        )
    } else if (!isFirst) {
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = theme.spacing.sm, vertical = theme.spacing.xs)
                .height(1.dp)
                .background(theme.colors.border),
        )
    }
    section.items.forEach { item ->
        NavRow(
            item = item,
            isChild = false,
            isSelected = selectedItemId == item.id,
            isExpanded = item.id in expandedGroupIds,
            onClick = {
                if (item.children.isNotEmpty()) onToggleGroup(item.id) else onSelect(item)
            },
            iconForItemId = iconForItemId,
        )
        AnimatedVisibility(visible = item.children.isNotEmpty() && item.id in expandedGroupIds) {
            Column {
                item.children.forEach { child ->
                    NavRow(
                        item = child,
                        isChild = true,
                        isSelected = selectedItemId == child.id,
                        isExpanded = false,
                        onClick = { onSelect(child) },
                        iconForItemId = iconForItemId,
                    )
                }
            }
        }
    }
}

@Composable
private fun NavRow(
    item: ShellNavItem,
    isChild: Boolean,
    isSelected: Boolean,
    isExpanded: Boolean,
    onClick: () -> Unit,
    iconForItemId: Map<String, ImageVector>,
) {
    val theme = LocalUiTheme.current
    val contentColor = if (isSelected) theme.colors.accent else theme.colors.textSecondary
    Row(
        verticalAlignment = Alignment.CenterVertically,
        modifier = Modifier
            .fillMaxWidth()
            .padding(start = if (isChild) theme.spacing.xl else 0.dp)
            .clip(RoundedCornerShape(theme.radius.md))
            .background(if (isSelected) theme.colors.accent.copy(alpha = 0.12f) else theme.colors.surface)
            .clickable(onClick = onClick)
            .padding(horizontal = theme.spacing.sm, vertical = theme.spacing.xs + 2.dp),
    ) {
        iconForItemId[item.id]?.let { icon ->
            Icon(
                imageVector = icon,
                contentDescription = null,
                tint = contentColor,
                modifier = Modifier
                    .size(20.dp)
                    .padding(end = 2.dp),
            )
            Spacer(modifier = Modifier.size(theme.spacing.sm))
        }
        Text(
            text = item.label,
            color = contentColor,
            fontSize = theme.typography.sizes.sm,
            fontWeight = FontWeight.SemiBold,
            maxLines = 1,
            modifier = Modifier.weight(1f),
        )
        val badgeText = item.badgeText
        if (!badgeText.isNullOrEmpty() && badgeText != "0") {
            Text(
                text = badgeText,
                color = theme.colors.accent,
                fontSize = theme.typography.sizes.xs,
                fontWeight = FontWeight.Bold,
                modifier = Modifier
                    .clip(RoundedCornerShape(theme.radius.pill))
                    .background(theme.colors.accent.copy(alpha = 0.14f))
                    .padding(horizontal = theme.spacing.xs, vertical = 1.dp),
            )
        }
        if (item.children.isNotEmpty()) {
            Icon(
                imageVector = Icons.Filled.KeyboardArrowDown,
                contentDescription = null,
                tint = theme.colors.textSecondary,
                modifier = Modifier
                    .size(16.dp)
                    .rotate(if (isExpanded) 180f else 0f),
            )
        }
    }
}

@Composable
private fun AccountFooter(profile: ShellProfile) {
    val theme = LocalUiTheme.current
    val context = LocalContext.current
    var isMenuOpen by remember { mutableStateOf(false) }

    Column(modifier = Modifier.padding(theme.spacing.sm)) {
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .height(1.dp)
                .background(theme.colors.border),
        )

        AnimatedVisibility(visible = isMenuOpen) {
            Column {
                val isLight = theme.mode == UiThemeMode.LIGHT
                profile.items.forEach { item ->
                    AccountMenuRow(
                        icon = item.icon ?: Icons.Filled.Settings,
                        label = item.label,
                        onClick = item.action,
                    )
                }
                AccountMenuRow(
                    icon = if (isLight) Icons.Filled.DarkMode else Icons.Filled.LightMode,
                    label = if (isLight) "Dark mode" else "Light mode",
                    onClick = { ThemeModeStore.setMode(context, if (isLight) "dark" else "light") },
                )
                profile.onSignOut?.let { onSignOut ->
                    AccountMenuRow(
                        icon = Icons.AutoMirrored.Filled.Logout,
                        label = "Sign out",
                        onClick = onSignOut,
                    )
                }
            }
        }

        Row(
            verticalAlignment = Alignment.CenterVertically,
            modifier = Modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(theme.radius.md))
                .clickable { isMenuOpen = !isMenuOpen }
                .padding(horizontal = theme.spacing.sm, vertical = theme.spacing.xs),
        ) {
            val avatarInitials: @Composable () -> Unit = {
                Box(
                    contentAlignment = Alignment.Center,
                    modifier = Modifier
                        .size(28.dp)
                        .clip(CircleShape)
                        .background(theme.colors.accent.copy(alpha = 0.14f)),
                ) {
                    Text(
                        text = initialsOf(profile.label),
                        color = theme.colors.accent,
                        fontSize = theme.typography.sizes.xs,
                        fontWeight = FontWeight.Bold,
                    )
                }
            }
            val avatarUrl = profile.avatarUrl
            if (avatarUrl != null) {
                RemoteAvatarImage(url = avatarUrl, size = 28.dp, placeholder = avatarInitials)
            } else {
                avatarInitials()
            }
            Column(
                modifier = Modifier
                    .weight(1f)
                    .padding(start = theme.spacing.sm),
            ) {
                Text(
                    text = profile.label,
                    color = theme.colors.textPrimary,
                    fontSize = theme.typography.sizes.sm,
                    fontWeight = FontWeight.SemiBold,
                    maxLines = 1,
                )
                profile.sublabel?.let { sublabel ->
                    Text(
                        text = sublabel,
                        color = theme.colors.textSecondary,
                        fontSize = theme.typography.sizes.xs,
                        maxLines = 1,
                    )
                }
            }
            Icon(
                imageVector = Icons.Filled.KeyboardArrowUp,
                contentDescription = null,
                tint = theme.colors.textSecondary,
                modifier = Modifier
                    .size(16.dp)
                    .rotate(if (isMenuOpen) 180f else 0f),
            )
        }
    }
}

@Composable
private fun AccountMenuRow(icon: ImageVector, label: String, onClick: () -> Unit) {
    val theme = LocalUiTheme.current
    Row(
        verticalAlignment = Alignment.CenterVertically,
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(theme.radius.md))
            .clickable(onClick = onClick)
            .padding(horizontal = theme.spacing.sm, vertical = theme.spacing.xs + 2.dp),
    ) {
        Icon(
            imageVector = icon,
            contentDescription = null,
            tint = theme.colors.textPrimary,
            modifier = Modifier.size(18.dp),
        )
        Text(
            text = label,
            color = theme.colors.textPrimary,
            fontSize = theme.typography.sizes.sm,
            fontWeight = FontWeight.SemiBold,
            modifier = Modifier.padding(start = theme.spacing.sm),
        )
    }
}

private fun initialsOf(label: String): String {
    val parts = label.split(' ', '@', '.', '_', '-').filter { it.isNotEmpty() }
    if (parts.isEmpty()) return "?"
    if (parts.size == 1) return parts[0].take(2).uppercase(Locale.getDefault())
    return "${parts[0].first()}${parts[1].first()}".uppercase(Locale.getDefault())
}
